libcaf_core/caf/uuid.test.cpp

```diff
@@ -0,0 +1,180 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/uuid.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/scenario.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/json_reader.hpp"
+#include "caf/json_writer.hpp"
+
+using namespace caf;
+
+namespace {
+
+uuid operator"" _uuid(const char* cstr, size_t cstr_len) {
+  if (cstr_len != 36)
+    test::runnable::current().fail("malformed test input");
```

Comment:
```
  auto& this_test = test::runnable::current();
  if (cstr_len != 36)
    this_test.fail("malformed test input");
```

(and use `this_test` below as well)

Reply:
I have refactored the tests a little bit for this change. Previously, the uuid's were being generated from the fixture and the  `runnable` object was not available at that point which causes the tests to fail. Moving the object declarations inside the test solves this issue. Please let me know if you have any feedback regarding this refactor.

Reply:
Good call, LGTM.